compiler/rustc_parse/src/parser/path.rs

```diff
@@ -381,7 +381,10 @@ impl<'a> Parser<'a> {
                     if self.may_recover()
                         && prev_token_before_parsing.kind == token::PathSep
                         && (style == PathStyle::Expr && self.token.can_begin_expr()
-                            || style == PathStyle::Pat && self.token.can_begin_pattern())
+                            || style == PathStyle::Pat
```

Comment:
This is uh... weird but also I have no idea what this recovery is doing, so I'm going to ignore it.